### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by esenharris4**
## Description

I added a new anchor route which allows for those with higher permissions to give route permissions to different users.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by esenharris4** [comment]
Completed the user and role permissions routes. Now we just have to create a system to send the json through

File: server/api/permissions.js [hunk 1]
```diff
@@ -106,6 +107,138 @@ const register = function (server, serverOptions) {
...
+        }
+      }]
+    },
+    handler: async function (request,h) {
+
+      const objectid = request.params.id;
+      const permissions = request.payload.permissions;
+      const name = request.payload.name;
+
+      return await Role.updateRole(objectid,permissions,name);
```
Comment (gregfrasco) [inline server/api/permissions.js]: Don't need to write a function within the roles model. Just call `await Role.findbyIdandUpdate(id, {$set {...}})`

File: server/api/permissions.js [hunk 2]
```diff
@@ -106,6 +107,138 @@ const register = function (server, serverOptions) {
...
+          return h.continue;
+        }
+      }]
+    },
+    handler: async function (request,h) {
+
+      const objectid = request.params.id;
+      const permissions = request.payload.permissions;
+
+      return await User.updateUserPermissions(objectid,permissions);
```
Comment (gregfrasco) [inline server/api/permissions.js]: Remove function, just update the user in this function

**Review by gregfrasco — CHANGES_REQUESTED** [review]

File: server/models/role.js [hunk 3]
```diff
@@ -25,6 +26,7 @@ class Role extends AnchorModel {
     return role[0];
 
   }
+
```
Comment (gregfrasco) [inline server/models/role.js:28]: remove

File: server/models/user.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const Assert = require('assert');
 const Bcrypt = require('bcrypt');
 const Hoek = require('hoek');
 const Joi = require('joi');
+const ObjectId = require('mongodb').ObjectID;
```
Comment (gregfrasco) [inline server/models/user.js]: remove

File: server/models/role.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ const Joi  = require('joi');
 const Assert = require('assert');
 const AnchorModel = require('../anchor/anchor-model');
 const Hoek = require('hoek');
+const ObjectId = require('mongodb').ObjectID;
```
Comment (gregfrasco) [inline server/models/role.js]: remove

**Review by gregfrasco — APPROVED** [review]

**Pull request merged**